### PR TITLE
Adding events consumer which provides an example of working with email events

### LINKED ISSUE #649: Create a Dockerized event webhook consumer
#### Issue Summary

SendGrid has an [Event Webhook](https://sendgrid.com/docs/API_Reference/Event_Webhook/event.html) which posts events related to your email activity to a URL of your choice. We want to have an easily deployable solution that allows for customers to easiy get up and running processing (parse and save) their event webhooks.

#### Acceptance Criteria

* A folder which contains a Dockerized solution
* Example documentation on how to deploy locally and start processing events
* Example documentation on how to deploy on at least one cloud service and start processing events
* An example complete payload with a test to verify the data is parsed properly

### PR DESCRIPTION
**Pull request by Taiters**
# Resolves #649 

### Checklist
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guide] and my PR follows them.
- [ ] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:

This PR adds
* A Dockerized Flask app which provides an example of working with email events
* A readme with guides for
    * Deploying locally with test data
    * Deploying locally with real data (using ngrok)
    * Deploying to Heroku

This repo already contains a Procfile for the email ingress example, so I've suggested copying the `events/` dir into a new repo on the users computer as part of the Heroku deploy to keep them separate.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=660) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=660) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=660) it.</sub>

**Comment by thinkingserious** [comment]
Hello @Taiters,

Could you please take a moment to sign our CLA so that we can merge this?

Also, could you please merge this against our [v4 branch](https://github.com/sendgrid/sendgrid-python/tree/v4)? Thank you!

With Best Regards,

Elmer

**Comment by thinkingserious** [comment]
Hello @Taiters,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by Taiters** [comment]
Hey @thinkingserious,

I've added my latest changes (docs and a slightly way to view events) and updated the base branch to `v4`.

Let me know if anything else needs added.

Thanks!
Dan

**Comment by Taiters** [comment]
Yeah well spotted @misterdorm! I've removed that folder from the PR

File: events/test.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import json
+import requests
+
+
+def read_events():
+    with open('sample_data.json', 'r') as f:
+        body = f.read()
+        return json.loads(body)
```
Comment (42B) [inline events/test.py]: Isn't this what [`json.load`](https://docs.python.org/3/library/json.html#json.load) is for?

```python
with open('sample_data.json', 'r') as f:
    return json.load(f)
```

**Comment by Taiters** [comment]
@42B yeah you're right. Changed in 84837b7035296ce0b4304312aa0702510d5839b1

**Comment by thinkingserious** [comment]
Hello @Taiters,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 15 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

**Pull request closed without merging**